marklogic-client-api/src/test/java/com/marklogic/client/test/rows/ValidateDocTest.java

```diff
@@ -136,49 +221,92 @@ public void testUsingFromParamAndSchematron() {
                 "</user>").withFormat(Format.XML));
         plan = plan.bindParam(param, new JacksonHandle(array), Collections.singletonMap("doc", attachments));
 
+
+        List<RowRecord> rows = resultRows(plan);
+        assertEquals(1, rows.size());
```

Comment:
@rjrudin is this test missing some binding? I get this error - 
Local message: failed to apply resource at rows: Bad Request. Server Message: OPTIC-INVALARGS: fn.error(null, 'OPTIC-INVALARGS', 'missing or empty URI value in write()'); -- Invalid arguments: missing or empty URI value in write()

Reply:
Yes - I recommend using `bindParam(String, DocumentWriteSet)` as shown in `xmlDocumentWriteSet` in `FromParamWriteTest`. I think that's the easiest way to setup a row with URI + doc + permissions. In fact, for writing documents, I think a user will always want to use a `DocumentWriteSet` or sequence of `DocumentWriteOperation`s because it's otherwise a bit of a pain to get the URI + doc + permissions columns together.